File: src/SIL.XForge.Scripture/ClientApp/src/app/core/models/sf-project-settings.ts
import { CheckingAnswerExport } from 'realtime-server/lib/esm/scriptureforge/models/checking-config';

/**
 * This interface represents the project settings that can be updated using "SFProjectService.onlineUpdateSettings()".
 */
export interface SFProjectSettings {
  translationSuggestionsEnabled?: boolean | null;
  sourceParatextId?: string | null;
  biblicalTermsEnabled?: boolean | null;

  /* DraftSourcesSettingsChange */
  additionalTrainingDataFiles?: string[] | null;
  draftingSourcesParatextIds?: string[] | null;
  trainingSourcesParatextIds?: string[] | null;

  checkingEnabled?: boolean | null;
  usersSeeEachOthersResponses?: boolean | null;
  checkingAnswerExport?: CheckingAnswerExport | null;
  hideCommunityCheckingText?: boolean | null;

  translatorsShareEnabled?: boolean | null;
  communityCheckersShareEnabled?: boolean | null;
  commentersShareEnabled?: boolean | null;
  viewersShareEnabled?: boolean | null;

  lynxAutoCorrectionsEnabled?: boolean | null;
  lynxAssessmentsEnabled?: boolean | null;
  lynxPunctuationCheckerEnabled?: boolean | null;
  lynxAllowedCharacterCheckerEnabled?: boolean | null;
}

File: src/SIL.XForge.Scripture/Models/SFProjectSettings.cs
using System;
using System.Collections.Generic;

namespace SIL.XForge.Scripture.Models;

/// <summary>
/// This class represents the project settings that can be updated using
/// <see cref="SIL.XForge.Scripture.Controllers.SFProjectsRpcController.UpdateSettings"/>.
/// </summary>
public class SFProjectSettings
{
    // translate settings
    public bool? TranslationSuggestionsEnabled { get; set; }
    public string? SourceParatextId { get; set; }
    public bool? BiblicalTermsEnabled { get; set; }
    public IEnumerable<string>? AdditionalTrainingDataFiles { get; set; }

    [Obsolete("For backwards compatibility with older frontend clients. Deprecated October 2025.")]
    public bool? AlternateSourceEnabled { get; set; }
    public IEnumerable<string>? DraftingSourcesParatextIds { get; set; }
    public IEnumerable<string>? TrainingSourcesParatextIds { get; set; }

    // checking settings
    public bool? CheckingEnabled { get; set; }
    public bool? UsersSeeEachOthersResponses { get; set; }
    public string? CheckingAnswerExport { get; set; }
    public bool? HideCommunityCheckingText { get; set; }

    // lynx settings
    public bool? LynxAutoCorrectionsEnabled { get; set; }
    public bool? LynxAssessmentsEnabled { get; set; }
    public bool? LynxPunctuationCheckerEnabled { get; set; }
    public bool? LynxAllowedCharacterCheckerEnabled { get; set; }
}
